### PR TITLE
Extract citations, use DocumenterCitations.jl

### LINKED ISSUE #152: references in docs
Would be nice if there was a way keep citations in a global bibtex file and have them rendered nicely in the docstrings.

See https://github.com/JuliaDocs/Documenter.jl/issues/379, [this discussion](https://discourse.julialang.org/t/citations-in-docs/5066/36) and the unregistered [Citations.jl](https://github.com/adamslc/Citations.jl) package.

### PR DESCRIPTION
**Pull request by charleskawczynski**
### Description

This PR extracts one citation (more to come!) and uses [DocumenterCitations.jl](https://github.com/ali-ramadhan/DocumenterCitations.jl) to cite it via a `.bib` file in `docs/`. The reference is included in `References.md`, which I added to the top-level of the docs page.

Closes #152!

We need to wait for Bibliography.jl to tag a new release because it currently has incompatible compact bounds. So, waiting on [#12](https://github.com/Azzaare/Bibliography.jl/issues/12)

- [x] Code follows the [style guidelines](https://clima.github.io/ClimateMachine.jl/latest/DevDocs/CodeStyle/) OR N/A.
- [x] Unit tests are included OR N/A.
- [x] Code is exercised in an integration test OR N/A.
- [x] Documentation has been added/updated OR N/A.

### CODE REVIEW

**Comment by ali-ramadhan** [comment]
Ah I was working on a Documenter.jl plugin for doing BibTeX citations/references (https://github.com/ali-ramadhan/DocumenterBibliographyTest.jl) but still need to flesh it out into a proper Documenter.jl plugin.

It's a little rough around the edges but it works. I'm currently using it to add citations/references to Oceananigans.jl documentation: https://clima.github.io/OceananigansDocumentation/stable/references/ (for citation examples: https://clima.github.io/OceananigansDocumentation/stable/verification/lid_driven_cavity/)

If you're interested we could maybe work on it together to get a plugin that works for both of us?

X-Ref: https://github.com/JuliaDocs/Documenter.jl/issues/1162

**Comment by charleskawczynski** [comment]
I'll revamp this PR once [DocumenterCitations](https://github.com/ali-ramadhan/DocumenterCitations.jl) is registered 🚀

**Comment by charleskawczynski** [comment]
Looks like we're getting close. Still need to work out dependency/compat issues to add `DocumenterCitations`.

**Comment by charleskawczynski** [comment]
Alright, looks like we're ready to go! @simonbyrne

**Review by simonbyrne — APPROVED** [review]

File: Project.toml [hunk 1]
```diff
@@ -50,29 +50,22 @@ WriteVTK = "64499a7a-5c06-52f2-abe2-ccb03c286192"
...
-Distributions = "0.22, 0.23, 0.24"
 DocStringExtensions = "0.8"
 DoubleFloats = "1.1"
 FFTW = "1.2"
 FileIO = "1.2"
 Formatting = "0.4"
 ForwardDiff = "0.10"
 GaussQuadrature = "0.5"
-JLD2 = "0.1, 0.2"
-KernelAbstractions = "0.4.1"
```
Comment (simonbyrne) [inline Project.toml:67]: actually, why have these all been removed?

Comment (charleskawczynski) [reply Project.toml:67]: I followed https://discourse.julialang.org/t/pkg-unsatisfiable-requirements-detected/22269/6

**Comment by simonbyrne** [comment]
If a83a793 doesn't work, we can revert it.

**Pull request merged**